workspaces/liferay-dlfoldertemplate-workspace/client-extensions/liferay-dlfoldertemplate-custom-element/src/components/template-list/controls/template-item-create-folder/template-item-create-folder.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {ClayInput} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {Flex, Form, TreeSelect} from 'antd';
+import React, {useCallback, useEffect, useState} from 'react';
+
+import {
+	getDocumentFolderDocumentFoldersPage,
+	getSiteDocumentFoldersPage,
+} from '../../../../services/folder-selector.service';
+import {createFolder} from '../../../../services/template-item-create-folder.service';
+import {ApplicationUtil} from '../../../../utils/appUtil';
+
+import './template-item-create-folder.css';
+
+const TemplateItemCreateFolder = ({templateID}) => {
+	const [folderTree, setFolderTree] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+	const [isSubmitting, setIsSubmitting] = useState(false);
+	const [form] = Form.useForm();
+
+	const loadFolderTree = async () => {
+		const loadSubFolder = async (folder) => {
+			const subFolders = await getDocumentFolderDocumentFoldersPage(
+				folder
+			);
+			const normalizedFolders = subFolders.items.map((folder) => ({
+				childrenCount: folder.numberOfDocumentFolders,
+				icon: 'folder',
+				id: Number(folder.id),
+				isLeaf: folder.numberOfDocumentFolders === 0,
+				key: folder.id,
+				label: folder.name,
+				title: folder.name,
+				type:
+					folder.numberOfDocumentFolders > 0
+						? 'repository'
+						: 'folder',
+				value: folder.id,
+			}));
+
+			return normalizedFolders;
+		};
+		const root = (
+			await getSiteDocumentFoldersPage(
+				ApplicationUtil.getLiferay().ThemeDisplay.getScopeGroupId()
+			)
+		).items.map((folder) => ({
+			childrenCount: folder.numberOfDocumentFolders,
+			id: Number(folder.id),
+			isLeaf: folder.childrenCount <= 0,
+			key: folder.id,
+			label: folder.name,
+			selected: false,
+			title: folder.name,
+			value: folder.id,
```

Comment:
This looks like another duplicate normalization process. Creating a shared function for this will be cleaner.

Reply:
actually its not the same process, i am using one to build the root folder and one for the subfolder, so its not the same process if i got your point in the right way
const root = (await getSiteDocumentFoldersPage(scopeGroupId)).items.map(
			(folder) => ({
				childrenCount: folder.numberOfDocumentFolders,
				id: Number(folder.id),
				isLeaf: folder.childrenCount <= 0,
				key: folder.id,
				label: folder.name,
				selected: false,
				title: folder.name,
				value: folder.id,
			})
		); 
		and 
		
		const normalizedFolders = subfolders.items.map((folder) => ({
				childrenCount: folder.numberOfDocumentFolders,
				icon: 'folder',
				id: Number(folder.id),
				isLeaf: folder.numberOfDocumentFolders === 0,
				key: folder.id,
				label: folder.name,
				title: folder.name,
				type:
					folder.numberOfDocumentFolders > 0
						? 'repository'
						: 'folder',
				value: folder.id,
			}));